### DISCUSSION TITLE
How to collapse by default the attributes section of a rich Dataset representation?

### DISCUSSION THREAD

**Discussion by eschalkargans** [pydata/xarray#9066] [Q&A] [answered]
Hello,

I wonder if there is maybe a xarray global option to collapse by default the attribute section in case of an extremely long attribute to prevent _attribute spamming_.

An example:

```python
xds = xr.Dataset({"variable_name": np.arange(10)}, attrs={"one_long_attr": "lorem ipsum " * 10000})
xds
```

![Screenshot from 2024-06-04 14-45-48](https://github.com/pydata/xarray/assets/119882363/bfe3111d-7e92-4565-be29-6ced0c9fd698)

![Screenshot from 2024-06-04 14-45-39](https://github.com/pydata/xarray/assets/119882363/262e0eb6-db95-41bd-8584-5edb480b3bb2)

The text-only version is immune against this issue as the attribute can only express itself in the space of a line:

```python
print(xds)
```

```
<xarray.Dataset>
Dimensions:        (variable_name: 10)
Coordinates:
  * variable_name  (variable_name) int64 0 1 2 3 4 5 6 7 8 9
Data variables:
    *empty*
Attributes:
    one_long_attr:  lorem ipsum lorem ipsum lorem ipsum lorem ipsum lorem ips...

```

:information_source: I already made an issue on the `datatree` repo, but this display issue is also about xarray: https://github.com/xarray-contrib/datatree/issues/332 ; if such a global option does not exist I can create an issue to add this option.

Thanks!

**Comment by keewis** [comment, marked as answer]
you can use
```python
xr.set_options(display_expand_attrs=False)
```
for that.

(see also the [documentation](https://docs.xarray.dev/en/stable/generated/xarray.set_options.html) of `xr.set_options` for a full list of available options)

**Comment by kmuehlbauer** [comment]
@eschalkargans please have a look here https://docs.xarray.dev/en/stable/user-guide/options.html

**Comment by eschalkargans** [comment]
Thanks!
